ipamd/ipamd.go

```diff
@@ -307,3 +317,89 @@ func (c *IPAMContext) nodeIPPoolTooHigh() bool {
 	return (total-used > 2*c.currentMaxAddrsPerENI)
 
 }
+
+// nodeIPPoolReconcile reconcile ENI and IP info from metadata service and IP addresses in datastore
+func (c *IPAMContext) nodeIPPoolReconcile() error {
+	log.Debug("Reconciling IP pool info...")
+	attachedENIs, err := c.awsClient.GetAttachedENIs()
+
+	if err != nil {
+		log.Error("ip pool reconcile: Failed to get attached eni info", err.Error())
+		errors.Wrap(err, "ip pool reconcile: failed to get attached eni")
+	}
+
+	curENIs := c.dataStore.GetENIInfos()
+
+	// mark phase
+	for _, attachedENI := range attachedENIs {
+		eniIPPool, err := c.dataStore.GetENIIPPools(attachedENI.ENIID)
+
+		if err == nil {
+			log.Debugf("Reconcile existing ENI %s IP pool", attachedENI.ENIID)
+			// reconcile IP pool
+			c.eniIPPoolReconcile(eniIPPool, attachedENI, attachedENI.ENIID)
+
+			// mark action= remove this eni from curENIs list
+			delete(curENIs.ENIIPPools, attachedENI.ENIID)
```

Comment:
I dont see any lock here, if Eni gets created just before GetEniInfos() call, isn't it will removed in sweep section?